ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitterSpec.scala

```diff
@@ -563,6 +565,84 @@ class TransactionCommitterSpec extends AnyWordSpec with Matchers with MockitoSug
       }.toMap
   }
 
+  "validateContractKeys" should {
+    "return Inconsistent error when a contract key resolves to a different contract id than given by a participant node" in {
```

Comment:
I guess you also want to test for the `None` vs. `Some` cases here.

Reply:
Still TODO but noted, will work on it shortly.

Reply:
Done.